src/Std/Time/Time/PlainTime.lean

```diff
@@ -30,213 +30,225 @@ structure PlainTime where
   /--
   `Second` component of the `PlainTime`
   -/
-  second : Sigma Second.Ordinal
+  second : Second.Ordinal leap
 
   /--
   `Nanoseconds` component of the `PlainTime`
   -/
   nanosecond : Nanosecond.Ordinal
   deriving Repr
 
-instance : Inhabited PlainTime where
-  default := ⟨0, 0, Sigma.mk false 0, 0, by decide⟩
+/--
+COMMENT
+-/
+def LeapTime := PlainTime True
+
+instance : Inhabited (PlainTime x) where
+  default := ⟨0, 0, 0, 0, by decide⟩
 
-instance : BEq PlainTime where
+instance : BEq (PlainTime x) where
   beq x y := x.hour.val == y.hour.val && x.minute == y.minute
-          && x.second.snd.val == y.second.snd.val && x.nanosecond == y.nanosecond
+          && x.second.val == y.second.val && x.nanosecond == y.nanosecond
 
 namespace PlainTime
 
 /--
 Creates a `PlainTime` value representing midnight (00:00:00.000000000).
 -/
-def midnight : PlainTime :=
-  ⟨0, 0, ⟨true, 0⟩, 0⟩
+def midnight : PlainTime α :=
```

Comment:
What is the rational behind using `α` and not `leap` for this parameter in these definitions? It seems that a name of `leap` would be better when users wish to set it using named parameters,